src/zap/zapinfo.cpp

```diff
@@ -1859,17 +1860,40 @@ PVOID ZapInfo::embedDirectCall(CORINFO_METHOD_HANDLE ftn,
     {
         if (!fAllowThunk)
         {
-            return NULL;
+            return FALSE;
         }
 
-        pEntryPointOrThunkToEmbed = m_pImage->GetImportTable()->GetExternalMethodThunk(ftn);
+        if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_ZapNoRWX))
```

Comment:
The conflig value checks are relatively slow. You may want to do this check once when we are initializating the sections, and then just use null check against the section pointer here.

Reply:
Fixed, thanks for the info.